riscv-runtime/src/commit.rs

```diff
@@ -0,0 +1,55 @@
+pub static mut PUBLICS: CommittedPublics = CommittedPublics::new();
+
+pub fn commit(n: u32) {
+    unsafe { PUBLICS.commit(n) }
+}
+
+pub(crate) fn finalize() -> [u64; 4] {
+    unsafe { PUBLICS.finalize() }
+}
+
+pub struct CommittedPublics {
+    state: [u64; 12],
+    buffer_size: u8,
+}
+
+impl CommittedPublics {
+    pub const fn new() -> Self {
+        Self {
+            state: [0; 12],
+            buffer_size: 0,
+        }
+    }
+
+    pub fn commit(&mut self, n: u32) {
+        self.state[self.buffer_size as usize + 4] = n as u64;
+
+        self.buffer_size += 1;
+
+        if self.buffer_size == 4 {
```

Comment:
Why 4?

Reply:
Every time we get 4 words we compute a hash, it's the same size as the output which becomes the input of the next round